src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -17,8 +17,14 @@ def validate(self, data):
             if scope not in valid_scopes:
                 raise ValidationError(u'{} not a valid scope'.format(scope))
 
+    def to_internal_value(self, data):
```

Comment:
Good question. It took me some time to wrap my head around this reading the docs as well.

The thing is that the methods are used to map between primitive and non-primitive data types (where dictionaries and lists are also considered primitive in this case).

`The to_internal_value() method is called to restore a primitive datatype into its internal python representation.`

In the case of `api_scopes`, the primitive data that crosses the wires will be a list of strings. And then when we use it on the server side, we'll still just need a list of strings.

One might then ask why we implement `to_internal_value` at all. That's because the framework requires that for `WritabelField`s:

`Note that the WritableField class that was present in version 2.x no longer exists. You should subclass Field and override to_internal_value() if the field supports data input.` (from the same doc).